bayem/vba.py

```diff
@@ -326,15 +338,25 @@ def run(self):
             self.c[n] = gamma.shape
 
         self.noise_groups = self.noise0.keys()
+        self.result.noise0 = self.p.original_noise(self.noise0)
+        self.result.noise0_dict = self.noise0
 
-        i_iter = 0
         while True:
-            i_iter += 1
-
             self.update_parameters(k, J)
-            
-            k, J = self.p(self.m)
-            
+            try:
+                k, J = self.p(self.m)
+                self.result.nit += 1
+            except self.options.allowed_exceptions as e:
+                _msg = (
+                    f"Some exception arised in the {self.result.nit+1}-th "
```

Comment:
That will likely look weird when it says "1th" or "23th" iteration and a remedy would be to say "... in iteration {nit} ...", but I don't really care :)